### PR TITLE
Add hardened ami

### PR DESCRIPTION
**Pull request by Dzero32**
Add's hardened CIS ami with the latest version form the marketplace

### CODE REVIEW

File: .github/workflows/publish-stack-set-to-service-catalog.yaml [hunk 1]
```diff
@@ -58,6 +58,23 @@ jobs:
...
+          LATEST-AMI=$(aws ec2 describe-images \
+          --owners aws-marketplace \
+          --filters 'Name=name,Values=*'${PRODUCT_ID}'*' \
+          --query 'Images[*].[ImageId,CreationDate]' --output text | sort -k2 -r | head -n1 | awk '{ print $1 }')
+          echo "amiId=$LATEST_AMI" >> $GITHUB_OUTPUT
+        env:
+          AWS_DEFAULT_REGION: eu-west-2
+          PRODUCT_ID: "c41d38c4-3f6a-4434-9a86-06dd331d3f9c"
       - name: cdk synth
         uses: youyo/aws-cdk-github-actions@v2
```
Comment (two-first-names) [inline .github/workflows/publish-stack-set-to-service-catalog.yaml:79]: Where does this come from?
Discuss if we should use this action.
Pin the action version, at the very least.

**Review by melvabout — APPROVED** [review]

File: .github/workflows/publish-stack-set-to-service-catalog.yaml [hunk 2]
```diff
@@ -66,18 +83,14 @@ jobs:
           working_dir: 'cdk.out'
           cdk_stack: DynatraceMonitoringRoleStack
```
Comment (two-first-names) [inline .github/workflows/publish-stack-set-to-service-catalog.yaml:84]: Should be DynatraceActivegateStack

Comment (two-first-names) [reply .github/workflows/publish-stack-set-to-service-catalog.yaml:84]: Also L114

**Review by two-first-names — CHANGES_REQUESTED** [review]

File: .github/workflows/publish-stack-set-to-service-catalog.yaml (hunk 2 above)
Comment (Dzero32) [reply .github/workflows/publish-stack-set-to-service-catalog.yaml:84]: @melvabout is this the case? As you made these changes previously... Happy to change just let me know :)

Comment (melvabout) [reply .github/workflows/publish-stack-set-to-service-catalog.yaml:84]: Name of the workflow, publish-stack-set-to-service-catalog.yaml doesn't reflect, what this is being used for anymore, based on as is I was going to strip it out as part of https://govukverify.atlassian.net/browse/OBS-87.  If you want to change it to DynatraceActivegateStack here and I will address DynatraceMonitoringRoleStack, DynatraceSSMProdStack and DynatraceSSMNonProdStack either in this workflow or alternates as part of OBS-87

**Pull request closed without merging**